File: .agents/skills/pagbank-connect/examples/code/order-pix/request.mjs
/**
 * Cria pedido PIX. Body: examples/requests/order-pix.json
 * PAGBANK_CONNECT_KEY=... node request.mjs
 */
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __dirname = dirname(fileURLToPath(import.meta.url));
const CONNECT_KEY = process.env.PAGBANK_CONNECT_KEY ?? (() => { throw new Error('Defina PAGBANK_CONNECT_KEY'); })();
const BASE_URL = 'https://ws.pbintegracoes.com/pspro/v7';

const body = JSON.parse(
  readFileSync(join(__dirname, '../../requests/order-pix.json'), 'utf8'),
);

const res = await fetch(`${BASE_URL}/connect/ws/orders`, {
  method: 'POST',
  headers: {
    Authorization: `Bearer ${CONNECT_KEY}`,
    'Content-Type': 'application/json',
    Platform: 'AI',
  },
  body: JSON.stringify(body),
});

const data = await res.json();
if (!res.ok) {
  console.error(data);
  process.exit(1);
}
// POST de pedido: sucesso esperado = HTTP 201
if (res.status !== 201) {
  console.warn('Aviso: esperado status 201, recebido', res.status);
}

console.log('Order ID:', data.id);
console.log('PIX text:', data.qr_codes?.[0]?.text?.slice(0, 80), '...');

File: .agents/skills/pagbank-connect/examples/code/order-pix/request.py
#!/usr/bin/env python3
"""Cria pedido PIX. Body: examples/requests/order-pix.json"""
import json
import os
from pathlib import Path

import requests

CONNECT_KEY = os.environ["PAGBANK_CONNECT_KEY"]
BASE_URL = "https://ws.pbintegracoes.com/pspro/v7"
ROOT = Path(__file__).resolve().parents[2]
body = json.loads((ROOT / "requests" / "order-pix.json").read_text())

response = requests.post(
    f"{BASE_URL}/connect/ws/orders",
    headers={
        "Authorization": f"Bearer {CONNECT_KEY}",
        "Content-Type": "application/json",
        "Platform": "AI",
    },
    json=body,
    timeout=60,
)
# POST de pedido: sucesso = HTTP 201 (requests trata 2xx como ok)
if not response.ok:
    print(response.status_code, response.text)
    response.raise_for_status()
assert response.status_code == 201, f"esperado 201, veio {response.status_code}"
data = response.json()
print("Order ID:", data.get("id"))
qr = (data.get("qr_codes") or [{}])[0]
print("PIX copy-paste:", qr.get("text", "")[:80], "...")
